src/Chapter1.hs

```diff
@@ -542,7 +543,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+  | x <= y && x >= z = x
+  | x <= z && x >= y = x
+  | y <= x && y >= z = y
+  | y <= z && y >= x = y
+  | z <= x && z >= y = z
+  | otherwise = z -- should be: z <= y && z >= x = z
```

Comment:
Yeap, you correctly noticed that.
Also, the previous case could be collapsed with the last one, as it doesn't matter that much, it still will return `z`:
```suggestion
  | otherwise = z -- should be: (z <= x && z >= y) || (z <= y && z >= x) = z
```